src/endpoint/s3/ops/s3_post_object_select.js

```diff
@@ -0,0 +1,97 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const dbg = require('../../../util/debug_module')(__filename);
+const S3Error = require('../s3_errors').S3Error;
+const s3_utils = require('../s3_utils');
+const nb_native = require('../../../util/nb_native');
+const stream_utils = require('../../../util/stream_utils');
+
+/**
+ * https://docs.aws.amazon.com/AmazonS3/latest/API/API_SelectObjectContent.html
+ */
+async function post_object_select(req, res) {
+
+    if (nb_native().S3Select === undefined) {
+        dbg.error("Request for object select but compiled without BUILD_S3SELECT=1");
+        res.statusCode = 501;
+        res.end();
+        return;
+    }
+
+    req.object_sdk.setup_abort_controller(req, res);
+    const agent_header = req.headers['user-agent'];
+    const noobaa_trigger_agent = agent_header && agent_header.includes('exec-env/NOOBAA_FUNCTION');
+    const encryption = s3_utils.parse_encryption(req);
+
+    const md_params = {
+        bucket: req.params.bucket,
+        key: req.params.key,
+        version_id: req.query.versionId,
+        encryption,
+    };
+    if (req.query.get_from_cache !== undefined) {
+        md_params.get_from_cache = true;
+    }
```

Comment:
This can be removed - this is a non standard query that we added to get object just for testing/troubleshooting of namespace_cache. If we ever decide to support this here we can add, but for now it seems irrelevant.